probe/endpoint/dns_snooper_linux_amd64.go

```diff
@@ -29,6 +30,14 @@ type DNSSnooper struct {
 	decodingErrorCounts map[string]uint64 // for limiting
 }
 
+type domainSet struct {
+	// Not worth using a RMutex since we don't expect a lot of contention
+	// and they are considerably larger (8 bytes vs 24 bytes), which would
+	// bloat the cache
+	sync.Mutex
```

Comment:
Very true, I initially created a global lock in the `DNSSnooper` (which is passed by pointer) and overlooked that detail when moving it here.

However, after thinking about I think it's fine (gcache doesn't copy, it uses interfaces, which hold a pointer).